### PR TITLE
Move the edit button in the site editor sidebar to a contextual widget

### PR DESCRIPTION
**Pull request by youknowriad**
Related #36667

## What?

This PR redesigns the sidebar of the browse mode to match the video shared here https://user-images.githubusercontent.com/846565/208665227-82537dfb-2893-4fb6-8f84-823f1e70cc07.mp4

## Why?

We have noticed that the "edit" button is a bit hidden and lacks context/connection to the frame. This PR adds a "widget" on top of the sidebar to add that missing context and move the edit button there for more clarity.

## How?

 - In order to so, I had to redesign the whole layout with a new structure. No more grid, now, it's all flex box
 - The header is fixed at the top
 - The island is fixed on top of the sidebar, that way it can morph into the header when moving to edit mode.

**Note**

 - At the moment, I didn't design the "title" yet, the piece of UI that show be showing the active template or template part title loaded in the frame. I'll follow-up with that quickly. I focused on the animation, positions...
 - In this initial PR, I'm not going to be adding any shortcuts to the header of the "edit" mode.

https://user-images.githubusercontent.com/272444/208882700-9cc76cb1-d1e2-4b72-915a-a2c1fdb6751a.mov

### CODE REVIEW

**Comment by jasmussen** [comment]
Thanks for this. I think there's a lot of sense in isolating the site information in a "hub" like that, letting the waterfall navigation slide below it. It is also a way, potentially, to allow for more context to show next to the Edit button in the future.

Can the edit button be 32px tall?

**Comment by jameskoster** [comment]
> Can the edit button be 32px tall?

Also the W logo / site icon for balance?

The W logo is currently 30x30, inside a 36px button, so will need some tweaking.

Edit: I noticed the sidebar seems to have shrunk a bit (width), and the top toolbar animation has disappeared.

Edit 2: One other small thing: When moving from browse to edit, the nav panel slides out to the left. Instead of sliding back in when moving from edit to browse it appears instantly. I don't have a strong preference re sliding vs fading, but they should probably be consistent?

**Comment by youknowriad** [comment]
> One other small thing: When moving from browse to edit, the nav panel slides out to the left. Instead of sliding back in when moving from edit to browse it appears instantly. I don't have a strong preference re sliding vs fading, but they should probably be consistent?

Unfortunately, this animation is controlled by a lower-level reusable component "Navigator", a fix might not be easy here. Maybe I can try to disable the sliding and only leave fading.

**Comment by youknowriad** [comment]
- I fixed the edit button height
- I fixed the sidebar width
- I fixed the top bar animation
-  Didn't manage to fix the exit animation of the sidebar content

How is this looking?

**Comment by jasmussen** [comment]
I think it's looking good!

anim

At the risk of repeating Jay on animation, the inserter button appearing "under" the hub animation feels awkward. Could we speed up the hub animation so it finishes before the white toolbar drops down? Or can we animate width of the white bar (might be awkward).

Conceptually, it makes sense that the hub is an element that floats _above_ the bottom dark gray canvas, and _above_ the white toolbar and inspector elements in elevation. That's already the case in this animation. But I think either speeding up the hub-collapsing animation might thread the needle. Alternately, maybe the contents of the top toolbar (all the buttons) could fade in as the toolbar drops down. That might also give the appearance that they aren't interactive yet.

**Comment by youknowriad** [comment]
The hub animates faster now.

File: packages/edit-site/src/components/layout/style.scss [hunk 1]
```diff
@@ -1,80 +1,102 @@
...
-.edit-site-layout__header {
-	grid-area: header;
-	height: $header-height;
+.edit-site-layout__hub {
+	position: fixed;
+	top: $canvas-padding;
+	left: $canvas-padding;
+	width: calc(100vw - #{$canvas-padding * 2});
+	height: $hub-height;
+	z-index: 3;
```
Comment (jorgefilipecosta) [inline packages/edit-site/src/components/layout/style.scss]: Should we remove this hardcoded value and create a z-index function value for this?

File: packages/edit-site/src/components/layout/style.scss [hunk 2]
```diff
@@ -1,80 +1,102 @@
...
+	}
 }
 
-.edit-site-layout__logo {
-	align-items: center;
+.edit-site-layout__header {
 	height: $header-height;
 	display: flex;
-	justify-content: space-between;
+	z-index: 2;
```
Comment (jorgefilipecosta) [inline packages/edit-site/src/components/layout/style.scss]: Same comment as above.

**Review by jorgefilipecosta — COMMENTED** [review]
Some end-to-end test failures may be relevant.

**Review by jasmussen — APPROVED** [review]
This feels like a solid upgrade over trunk, especially if paired with #46736, so greenlighting it as such:

status

I still think we need to make the nav sidebar 360px wide, and I wish there was a way we could gate the browse view mostly as an experiment, but since that's not really feasible. at least his increases contextuality and usability a fair bit.

**Review by jorgefilipecosta — APPROVED** [review]
Worked well on my tests 👍 And the code looks good.

**Pull request merged**